### PR TITLE
[14729] Remove pinned messages when unpin

### LINKED ISSUE #14729: It is possible to pin already pinned message
**Steps:**

1. Open any chat
2. Send a message
3. Pin that message
4. Close chat
5. Re enter the chat
6. Open list of pinned messages by tapping on pinned banner
7. Longtap on pinned message
8. See if unpin option is displayed for that message

**Actual result:** pin instead of unpin option is shown. It is possible perform pin action for already pinned message

**Expected result:** unpin option is displayed

https://user-images.githubusercontent.com/97245802/211273075-cb94c1d8-7d3c-47b0-8506-1e0bc64d00ca.mp4

### Additional Information

- Status version: nightly 
- Operating System: Android, iOS

[comment]: # (Please, add logs/notes if necessary)

### PR DESCRIPTION
**Pull request by ibrkhalil**
fixes #14729 
status: ready

### CODE REVIEW

**Comment by ibrkhalil** [comment]
Hey team
So in this PR I am trying to delete a message of type system message and I get an exception from status-go that we can't delete system message .. Can we add an exception for the case of pinned messages ?
```clj ERROR [status-im2.contexts.chat.messages.delete-message.events:126] - failed to delete message  {:message-id "0x19b7a900d4027bb1f3eb12d8679d717e15163d68dbe306436eced24416a4d622", :error {:code -32000, :message "message type cannot be deleted"}}```
image

CC: @status-im/clojure

And what is the desired look when the user deletes a pinned message ? @pedro-et

**Comment by flexsurfer** [comment]
cc @yqrashawn

**Comment by yqrashawn** [comment]
~There's no conclusion about this last time we discuss this with design team.~ There was conclusion, see below comment

The key point is what happens when USERA (with no permission to unpin message) delete a pinned message which posted by USERA but pinned by USERB (with permission to pin message).
Any updates? @pedro-et @xAlisher @John-44 
I believe this does affect what it will look like after pinned message got deleted.

Moreover, **please note.**
The current implementation of delete pinned message is unpin+delete without checking if user has the permission to unpin.
So status-go may complain about no permission to unpin when such thing happens.

So we need to
1. make sure what happens after delete pinned message
2. determine if we need to remove a "message pinned" system message or add a "message unpinned" or "pinned message deleted"

**Comment by John-44** [comment]
> There's no conclusion about this last time we discuss this with design team.
> 
> The key point is what happens when USERA (with no permission to unpin message) delete a pinned message which posted by USERA but pinned by USERB (with permission to pin message). Any updates? @pedro-et @xAlisher @John-44 I believe this does affect what it will look like after pinned message got deleted.
> 
> Moreover, **please note.** The current implementation of delete pinned message is unpin+delete without checking if user has the permission to unpin. So status-go may complain about no permission to unpin when such thing happens.
> 
> So we need to
> 
> 1. make sure what happens after delete pinned message
> 2. determine if we need to remove a "message pinned" system message or add a "message unpinned" or "pinned message deleted"

Message should be deleted everywhere, and if the message is pinned, the message should remain pinned but the message text should be replaced with "Message deleted"

File: src/status_im2/contexts/chat/messages/pin/events.cljs [hunk 1]
```diff
@@ -56,6 +57,24 @@
                  (assoc-in [:pin-message-lists chat-id]
                            (message-list/add-many nil (vals all-messages))))}))))
 
+(rf/defn delete-pinned-message
```
Comment (yqrashawn) [inline src/status_im2/contexts/chat/messages/pin/events.cljs]: I thought there's a `:pin-message/send-pin-message` event to unpin message
example at https://github.com/status-im/status-mobile/blob/637dc4e279da9e8d97fa9a937adce88b0facaf47/src/status_im/ui2/screens/chat/messages/message.cljs#L283

Is this not working?

Last time when I reproduced issue #14729 while working on delete message

I remember the issue is when re-enter the chat view, mobile will load message from status-go and the `:pinned` or another related key turns to `false`
which caused the issue

**Review by cammellos — APPROVED** [review]

**Comment by yqrashawn** [comment]
https://user-images.githubusercontent.com/15090582/216579715-c0965db2-a5f3-4418-9bdc-b8b9a2fe5300.mp4

seems still able to repin already pinned message after reenter chat view

**Comment by pavloburykh** [comment]
@ibrkhalil hi! Is this PR ready for testing or WIP? Please, move PR to E2E column as soon as it will be ready for testing. Thanx!

**Comment by ibrkhalil** [comment]
> CleanShot.2023-02-03.at.18.34.55.mp4 
> seems still able to repin already pinned message after reenter chat view

Fixed

**Review by yqrashawn — APPROVED** [review]

**Comment by pavloburykh** [comment]
@ibrkhalil thanx for the PR. Please, take a look at the following issue

### ISSUE 1 Cannot unpin message after chat reopening if pinned bottom sheet has been previously opened


**Steps:**

1. Open chat
2. Send a message
3. Pin message
4. Open bottom sheet with pinned messages
6. Close bottom sheet
7. Close the chat
8. Reopen the chat
9. Open bottom sheet with pinned messages
10. Longtap on pinned message
11. Tap unpin message
12. See if message has been unpinned

**Actual result:** message has not been unpinned. In fact pin of already pinned message action is performed. If to try unpinning message once again - it works. 

**Expected result:** message has been unpinned. 


https://user-images.githubusercontent.com/97245802/216988942-8307e9fd-1bce-4151-ae0a-e26c2f1419f5.mp4

**Comment by ibrkhalil** [comment]
> @ibrkhalil thanx for the PR. Please, take a look at the following issue
> 
> ### ISSUE 1 Cannot unpin message after chat reopening if pinned bottom sheet has been previously opened
> **Steps:**
> 
> 1. Open chat
> 2. Send a message
> 3. Pin message
> 4. Open bottom sheet with pinned messages
> 5. Close bottom sheet
> 6. Close the chat
> 7. Reopen the chat
> 8. Open bottom sheet with pinned messages
> 9. Longtap on pinned message
> 10. Tap unpin message
> 11. See if message has been unpinned
> 
> **Actual result:** message has not been unpinned. In fact pin of already pinned message action is performed. If to try unpinning message once again - it works.
> 
> **Expected result:** message has been unpinned.
> 
>  telegram-cloud-document-2-5406581201018303568.mp4

Kindly check again Pavlo!
Thanks

**Comment by pavloburykh** [comment]
> Kindly check again Pavlo! Thanks

Hi @ibrkhalil ! ISSUE 1 is still reproducible in case with 3 pinned messages in the list. 

**Steps:**
1. Start a chat
2. Pin 3 messages to reach pin limit
3. Open bottom sheet with pinned messages
4. Close bottom sheet
5. Close the chat
6. Reopen the chat
7. Open bottom sheet with pinned messages
8. Longtap on pinned message
9. Tap unpin message
10. See if message has been unpinned

**Actual result:** it is impossible to unpin message as pin instead of unpin action is performed. User observes the warning about pin messages limit. 

**Expected result:** message is unpinned


https://user-images.githubusercontent.com/97245802/217506270-90fbb688-8f8a-4a98-b9a0-d08448e18cc7.mp4

**Comment by pavloburykh** [comment]
@ibrkhalil hi! Could you please take a look at the above issue. We need to include current PR in release cut which is currently in process of testing. Thanx.

**Comment by ibrkhalil** [comment]
> @ibrkhalil hi! Could you please take a look at the above issue. We need to include current PR in release cut which is currently in process of testing. Thanx.

Sorry Pavlo, Forgot to tell you it's ready to be re-tested .. Fixing the build now
UPDATE: Build fixed, Kindly take a look

**Comment by pavloburykh** [comment]
> Build fixed, Kindly take a look

thanx @ibrkhalil ! PR is ready to be merged into develop. For now we decided not to include this fix in release branch.

**Pull request merged**